atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -194,19 +196,22 @@ static int adjustBatchParameter(int parameterValue) {
 
     private Optional<TableToSweep> getTableToSweep() {
         return specificTableSweeper.getTxManager().runTaskWithRetry(
-                tx -> {
-                    Optional<SweepProgress> progress = specificTableSweeper.getSweepProgressStore().loadProgress(
-                            tx);
-                    if (progress.isPresent()) {
-                        return Optional.of(new TableToSweep(progress.get().tableRef(), progress.get()));
-                    } else {
-                        Optional<TableReference> nextTable = nextTableToSweepProvider.chooseNextTableToSweep(
-                                tx, specificTableSweeper.getSweepRunner().getConservativeSweepTimestamp());
-                        if (nextTable.isPresent()) {
-                            log.debug("Now starting to sweep next table.", UnsafeArg.of("table name", nextTable));
-                            return Optional.of(new TableToSweep(nextTable.get(), null));
+                new TransactionTask<Optional<TableToSweep>, RuntimeException>() {
```

Comment:
It's worth adding a comment, in case the next person feels like making the same refactor (I think you're already the second).